src/ai/backend/agent/docker/agent.py

```diff
@@ -1501,18 +1548,35 @@ async def handle_action_die(kernel_id: KernelId, evdata: Mapping[str, Any]) -> N
                                     evdata["Action"],
                                     evdata["Actor"],
                                 )
-                            if evdata["Action"] == "start":
-                                await asyncio.shield(
-                                    self.docker_ptask_group.create_task(
-                                        handle_action_start(kernel_id, evdata),
+                            match evdata["Action"]:
+                                case "start":
+                                    await asyncio.shield(
+                                        self.docker_ptask_group.create_task(
+                                            handle_action_start(kernel_id, evdata),
+                                        )
                                     )
-                                )
-                            elif evdata["Action"] == "die":
-                                await asyncio.shield(
-                                    self.docker_ptask_group.create_task(
-                                        handle_action_die(kernel_id, evdata),
+                                    break
```

Comment:
I remember that Python's match syntax does not 'fall-through' to the subsequent case blocks. Please remove these unnecessary `break` statements, unless you are intending to exit the entire subscription loop!